Tasks/ServiceFabricUpdateAppVersions/Update-ServiceVersions.psm1

```diff
@@ -45,11 +45,11 @@
         $versionPrefix = $newManifest.ServiceManifest.Version
         $newVersion = $versionPrefix + $VersionSuffix
 
-        $newPackagesXml = @(
-            $newManifest.ServiceManifest.CodePackage,
-            $newManifest.ServiceManifest.ConfigPackage,
-            $newManifest.ServiceManifest.DataPackage
-        ) | Where { $_ }
+        $newPackagesXml = @()
+        $newPackagesXml += $newManifest.ServiceManifest.CodePackage
+        $newPackagesXml += $newManifest.ServiceManifest.ConfigPackage
+        $newPackagesXml += $newManifest.ServiceManifest.DataPackage
+        $newPackagesXml = $newPackagesXml | Where { $_ }
```

Comment:
'$newManifest.ServiceManifest.CodePackage', etc. are technically arrays. The most common case is for each array to have one element, which PowerShell was automatically 'expanding' so that we ended up with something like { A, B C }. However, if it had more than one element, it resulted in an array like this: { A, B, { C, D } } when really we wanted { A, B, C, D }

By using the '+=' operator, we get array concatenation rather than nested arrays